exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzureConfigProvider.java

```diff
@@ -0,0 +1,147 @@
+package com.netflix.exhibitor.core.config.azure;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.BlobProperties;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.netflix.exhibitor.core.azure.*;
+import com.netflix.exhibitor.core.config.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Properties;
+
+public class AzureConfigProvider implements ConfigProvider {
+    private final AzureConfigArguments arguments;
+    private final AzureClient azureClient;
+    private final String hostname;
+    private final Properties defaults;
+
+    /**
+     * @param factory     the factory
+     * @param credentials credentials
+     * @param arguments   args
+     * @param hostname    this VM's hostname
+     * @throws Exception errors
+     */
+    public AzureConfigProvider(AzureClientFactory factory, AzureCredential credentials, AzureConfigArguments arguments, String hostname) throws Exception {
+        this(factory, credentials, arguments, hostname, new Properties());
+    }
+
+    /**
+     * @param factory     the factory
+     * @param credentials credentials
+     * @param arguments   args
+     * @param hostname    this VM's hostname
+     * @param defaults    default props
+     * @throws Exception errors
+     */
+    public AzureConfigProvider(AzureClientFactory factory, AzureCredential credentials, AzureConfigArguments arguments, String hostname, Properties defaults) throws Exception {
+        this.arguments = arguments;
+        this.hostname = hostname;
+        this.defaults = defaults;
+        azureClient = factory.makeNewClient(credentials);
+    }
+
+    public AzureClient getAzureClient() {
+        return azureClient;
+    }
+
+    @Override
+    public void start() throws Exception {
+        // NOP
+    }
+
+    @Override
+    public void close() throws IOException {
+        azureClient.close();
+    }
+
+    @Override
+    public PseudoLock newPseudoLock() throws Exception {
+        return new AzurePseudoLock
+                (
+                        azureClient,
+                        arguments.getContainer(),
+                        arguments.getLockArguments().getPrefix(),
```

Comment:
matter of taste, really, but I'd create a temp variable:
    LockArguments arguments.getLockArguments();
and then use it here.
Either the compiler or the JVM may optimize all this away, but why take the chance?